packages/nodes-base/nodes/Google/CloudStorage/BucketDescription.ts

```diff
@@ -0,0 +1,617 @@
+								const filters = this.getNodeParameter('getFilters') as IDataObject;
+
+								// Merge in the options into the queryset
+								requestOptions.qs = Object.assign(requestOptions.qs, additionalQs, filters);
+								return requestOptions;
+							},
+						],
+					},
+				},
+				action: 'Create a new Bucket',
+			},
+		],
+		default: 'getAll',
+	},
+];
+
+export const bucketFields: INodeProperties[] = [
+	{
+		displayName: 'Project ID',
+		name: 'projectId',
+		type: 'string',
+		required: true,
+		placeholder: 'Project ID',
+		displayOptions: {
+			show: {
+				resource: ['bucket'],
+				operation: ['create', 'getAll'],
+			},
+		},
+		default: '',
+	},
+	{
+		displayName: 'Bucket Name',
+		name: 'bucketName',
+		type: 'string',
+		placeholder: 'Bucket Name',
+		required: true,
+		displayOptions: {
+			show: {
+				resource: ['bucket'],
+				operation: ['create', 'get', 'update', 'delete'],
+			},
+		},
+		default: '',
+	},
+	{
+		displayName: 'Prefix',
+		name: 'prefix',
+		type: 'string',
+		placeholder: 'Filter for Bucket Names',
+		displayOptions: {
+			show: {
+				resource: ['bucket'],
+				operation: ['getAll'],
+			},
+		},
+		default: '',
+	},
+	{
+		displayName: 'Projection',
+		name: 'projection',
+		type: 'options',
+		noDataExpression: true,
+		options: [
+			{
+				name: 'All Properties',
+				value: 'full',
+			},
+			{
+				name: 'No ACL',
+				value: 'noAcl',
+			},
+		],
+		default: 'noAcl',
+		displayOptions: {
+			show: {
+				resource: ['bucket'],
+				operation: ['create', 'get', 'getAll', 'update'],
+			},
+		},
+	},
+	{
+		displayName: 'Return All',
+		name: 'returnAll',
+		type: 'boolean',
+		displayOptions: {
+			show: {
+				resource: ['bucket'],
+				operation: ['getAll'],
+			},
+		},
+		default: false,
+		description: 'Whether to return all results or only up to a given limit',
+	},
+	{
+		displayName: 'Filters',
+		name: 'getFilters',
+		type: 'collection',
+		displayOptions: {
+			show: {
+				resource: ['bucket'],
+				operation: ['delete', 'get', 'update'],
+			},
+		},
+		default: {},
+		placeholder: 'Add Filter',
+		options: [
+			{
+				displayName: 'Metageneration Match',
+				name: 'ifMetagenerationMatch',
+				type: 'number',
+				description:
+					'Only return data if the metageneration value of the Bucket matches the sent value',
+				default: 0,
+			},
+			{
+				displayName: 'Metageneration Exclude',
+				name: 'ifMetagenerationNotMatch',
+				type: 'number',
+				description:
+					'Only return data if the metageneration value of the Bucket does not match the sent value',
+				default: 0,
+			},
+		],
+	},
+	{
+		displayName: 'Predefined Access Control',
+		name: 'createAcl',
+		type: 'collection',
+		noDataExpression: true,
+		default: {},
+		placeholder: 'Add Access Control Parameters',
+		displayOptions: {
+			show: {
+				resource: ['bucket'],
+				operation: ['create', 'update'],
+			},
+		},
+		options: [
+			{
+				displayName: 'Predefined ACL',
```

Comment:
We can add a `routing` block to this option, allowing us to effectively remove the `preSend` blocks above from create and update operations.
It should look like
```
				routing: {
					request: {
						qs: {
							predefinedAcl: '={{$value}}',
						}
					}
				},
```

Reply:
Oh yeah good point! The preSend functions for create / update were bigger but because I took out the common JSON body parsing I didn't actually realise that it had simplified down to this, thank you!

Reply:
Oh but I just realised, the `createAcl` is a collection so it doesn't guarantee having either `predefinedAcl` or `predefinedObjectAcl` defined for any request so shouldn't it stay in this function? :thinking:

Reply:
Oh wait I just got it never mind